File: Binary Tree Java/Invert Binary Tree/RecursiveSolution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root==null)
            return root;
        
        //Swapping left and right
        TreeNode left=root.left;
        root.left=root.right;
        root.right=left;
        
        //Repeating via recursion for subtrees
        invertTree(root.left);
        invertTree(root.right);
        
        return root;
    }
}

File: Binary Tree Java/Invert Binary Tree/RecursiveSolution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun invertTree(root: TreeNode?): TreeNode? {
        root?: return null
        
        //Swapping
        var root1 = root
        val left = root.left
        root1.left = root.right
        root1.right=left
        
        invertTree(root.left)
        invertTree(root.right)
        
        return root
    }
}
